test/services/remediations_api_test.rb

```diff
@@ -6,6 +6,8 @@
 class RemediationsAPITest < ActiveSupport::TestCase
   setup do
     @rule = rules(:one)
+    # Append non-canonical profile to the rule
+    profiles(:one).update!(parent_profile: profiles(:two))
```

Comment:
This case represents a tailored profile which adds a rule that does not exist in any other profile, which is something that can happen, but it's a more special case of the broader issue of a rule not having a profile.